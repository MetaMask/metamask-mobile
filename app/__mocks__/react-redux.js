export * from 'react-redux-test';
