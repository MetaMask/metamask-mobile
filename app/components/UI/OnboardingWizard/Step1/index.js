import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  View,
  Text,
  StyleSheet,
  InteractionManager,
  Platform,
} from 'react-native';
import Coachmark from '../Coachmark';
import Device from '../../../../util/device';
import setOnboardingWizardStep from '../../../../actions/wizard';
import { strings } from '../../../../../locales/i18n';
import onboardingStyles from './../styles';
import {
  MetaMetricsEvents,
  ONBOARDING_WIZARD_STEP_DESCRIPTION,
} from '../../../../core/Analytics';
import AnalyticsV2 from '../../../../util/analyticsV2';

import { ThemeContext, mockTheme } from '../../../../util/theme';
import generateTestId from '../../../../../wdio/utils/generateTestId';
import { ONBOARDING_WIZARD_STEP_1_CONTAINER_ID } from '../../../../../wdio/screen-objects/testIDs/Components/OnboardingWizard.testIds';

const styles = StyleSheet.create({
  main: {
    flex: 1,
  },
  coachmark: {
    marginHorizontal: 16,
  },
  coachmarkContainer: {
    flex: 1,
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: Device.isIphoneX() ? 36 : Device.isIos() ? 16 : 36,
  },
});

class Step1 extends PureComponent {
  static propTypes = {
    /**
     * Callback called when closing step
     */
    onClose: PropTypes.func,
    /**
     * Dispatch set onboarding wizard step
     */
    setOnboardingWizardStep: PropTypes.func,
  };

  /**
   * Dispatches 'setOnboardingWizardStep' with next step
   */
  onNext = () => {
    const { setOnboardingWizardStep } = this.props;
    setOnboardingWizardStep && setOnboardingWizardStep(2);
    InteractionManager.runAfterInteractions(() => {
      AnalyticsV2.trackEvent(MetaMetricsEvents.ONBOARDING_TOUR_STARTED, {
        tutorial_step_count: 1,
        tutorial_step_name: ONBOARDING_WIZARD_STEP_DESCRIPTION[1],
      });
    });
  };

  /**
   * Calls props 'onClose'
   */
  onClose = () => {
    const { onClose } = this.props;
    onClose && onClose(false);
  };

  /**
   * Returns content for this step
   */
  content = () => {
    const colors = this.context.colors || mockTheme.colors;
    const dynamicOnboardingStyles = onboardingStyles(colors);

    return (
      <View style={dynamicOnboardingStyles.contentContainer}>
        <Text style={dynamicOnboardingStyles.content}>
          {strings('onboarding_wizard.step1.content1')}
        </Text>
        <Text style={dynamicOnboardingStyles.content}>
          {strings('onboarding_wizard.step1.content2')}
        </Text>
      </View>
    );
  };

  render() {
    return (
      <View
        style={styles.main}
        {...generateTestId(Platform, ONBOARDING_WIZARD_STEP_1_CONTAINER_ID)}
      >
        <View style={styles.coachmarkContainer}>
          <Coachmark
            title={strings('onboarding_wizard.step1.title')}
            content={this.content()}
            onNext={this.onNext}
            onBack={this.onClose}
            coachmarkStyle={styles.coachmark}
            action
          />
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  setOnboardingWizardStep: (step) => dispatch(setOnboardingWizardStep(step)),
});

Step1.contextType = ThemeContext;

export default connect(null, mapDispatchToProps)(Step1);
