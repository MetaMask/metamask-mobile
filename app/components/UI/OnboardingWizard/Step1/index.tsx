import React, { useCallback, useContext } from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import { useDispatch } from 'react-redux';

import Coachmark from '../Coachmark';
import Device from '../../../../util/device';
import setOnboardingWizardStep from '../../../../actions/wizard';
import { strings } from '../../../../../locales/i18n';
import onboardingStyles from '../styles';
import {
  MetaMetricsEvents,
  ONBOARDING_WIZARD_STEP_DESCRIPTION,
} from '../../../../core/Analytics';
import { ThemeContext, mockTheme } from '../../../../util/theme';
import generateTestId from '../../../../../wdio/utils/generateTestId';
import { ONBOARDING_WIZARD_STEP_1_CONTAINER_ID } from '../../../../../wdio/screen-objects/testIDs/Components/OnboardingWizard.testIds';
import { useMetrics } from '../../../../components/hooks/useMetrics';

const styles = StyleSheet.create({
  main: {
    flex: 1,
  },
  coachmark: {
    marginHorizontal: 16,
  },
  coachmarkContainer: {
    flex: 1,
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: Device.isIphoneX() ? 80 : Device.isIos() ? 40 : 64,
  },
});
interface Step1Props {
  onClose: () => Promise<void>;
}

const Step1 = ({ onClose }: Step1Props) => {
  const theme = useContext(ThemeContext) || mockTheme;
  const dynamicOnboardingStyles = onboardingStyles(theme.colors);
  const dispatch = useDispatch();
  const { trackEvent } = useMetrics();

  const content = useCallback(
    () => (
      <View style={dynamicOnboardingStyles.contentContainer}>
        <Text style={dynamicOnboardingStyles.content}>
          {strings('onboarding_wizard_new.step1.content1')}
        </Text>
      </View>
    ),
    [dynamicOnboardingStyles],
  );

  const onNext = useCallback(() => {
    dispatch(setOnboardingWizardStep?.(2));

    trackEvent(MetaMetricsEvents.ONBOARDING_TOUR_STARTED, {
      tutorial_step_count: 1,
      tutorial_step_name: ONBOARDING_WIZARD_STEP_DESCRIPTION[1],
    });
  }, [dispatch, trackEvent]);

  return (
    <View
      style={styles.main}
      {...generateTestId(Platform, ONBOARDING_WIZARD_STEP_1_CONTAINER_ID)}
    >
      <View style={styles.coachmarkContainer}>
        <Coachmark
          title={strings('onboarding_wizard_new.step1.title')}
          content={content()}
          onNext={onNext}
          onBack={onClose}
          coachmarkStyle={styles.coachmark}
          bottomIndicatorPosition={'bottomLeftCorner'}
          action
          onClose={onClose}
        />
      </View>
    </View>
  );
};

export default Step1;
