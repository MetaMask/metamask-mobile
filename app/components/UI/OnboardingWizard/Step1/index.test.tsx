import React from 'react';
import { shallow } from 'enzyme';
import Step1 from './';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Step1', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Step1 />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
