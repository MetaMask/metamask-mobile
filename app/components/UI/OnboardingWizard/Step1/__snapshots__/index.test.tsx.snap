// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1 should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="onboarding-wizard-step1-container-id"
>
  <View
    style={
      {
        "bottom": 80,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "opacity": 0,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#0376C9",
              "borderRadius": 8,
              "padding": 20,
            },
            {
              "marginHorizontal": 16,
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View />
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "fontFamily": "EuclidCircularB-Bold",
                "fontSize": 18,
                "fontWeight": "600",
              }
            }
          >
            Welcome to your wallet!
          </Text>
          <TouchableOpacity
            accessible={true}
            activeOpacity={1}
            disabled={false}
            onPress={[MockFunction]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "borderRadius": 8,
                "height": 16,
                "justifyContent": "center",
                "opacity": 1,
                "width": 16,
              }
            }
          >
            <SvgMock
              color="#FFFFFF"
              height={20}
              name="Close"
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              width={20}
            />
          </TouchableOpacity>
        </View>
        <View
          style={
            {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 20,
                "textAlign": "left",
              }
            }
          >
            On the blockchain, you need a wallet (and maybe some ETH). So letâ€™s take a look at how to use your MetaMask wallet.
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#0376C9",
                "borderColor": "#FFFFFF",
                "borderRadius": 16,
                "borderWidth": 1,
                "flex": 0.5,
                "flexDirection": "row",
                "height": 32,
                "justifyContent": "center",
                "marginRight": 4,
                "paddingHorizontal": 16,
              }
            }
            testID="onboarding-wizard-no-thanks-button"
            variant="Primary"
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Euclid Circular B",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              No thanks
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#0376C9",
                "borderRadius": 16,
                "borderWidth": 1,
                "flex": 0.5,
                "flexDirection": "row",
                "height": 32,
                "justifyContent": "center",
                "marginLeft": 4,
                "paddingHorizontal": 16,
              }
            }
            testID="onboarding-wizard-next-button"
            variant="Secondary"
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#0376C9",
                  "fontFamily": "Euclid Circular B",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Take the tour
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "marginBottom": 10,
            "marginLeft": 30,
            "top": -2,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderLeftColor": "transparent",
              "borderLeftWidth": 15,
              "borderRightColor": "transparent",
              "borderRightWidth": 15,
              "borderStyle": "solid",
              "borderTopColor": "#0376C9",
              "borderTopWidth": 12,
              "height": 0,
              "position": "absolute",
              "width": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
